BK-569 add AddressField component

jfplataroti opened the pull request:
Add AddressField component to be able to query locations with suggestions styled as the mockups.

To do the google autocomplete it uses: [React Google Places Autocomplete | React Google Places Autocomplete].

## Code changes
* Added `AddressField` as a wrapper for `react-google-places-autocomplete`.
* Added `AddressFieldOption` to be able to customize the autocomplete suggestions with our custom design.
* Added `AddressFieldContainer` to be able to add a prefix icon in the text input when it's needed.
* Added `AddressField.customization.ts` to add the custom theme for `react-select` used by `react-google-places-autocomplete`.

## Example

```tsx
<Form>
      <FormField>
        <AddressField placeholder="add address" icon={<Search />} />
      </FormField>
      ...
</Form>
```

![address-field](https://user-images.githubusercontent.com/1827213/105764184-a49ac580-5f24-11eb-8ed0-1a4c55c2e3c0.gif)

jfplataroti commented on packages/web/lib/theme.tsx:
```diff
@@ -2,7 +2,7 @@ import { grommet } from "grommet/themes";
 import { colors } from "grommet/themes/base";
 
-const BRAND_COLORS = {
+export const BRAND_COLORS = {
```
needed to export them to customize the styles of the `react-select`.

djoyal replied:
I think we want to avoid exporting this `const`. Instead we can use the theme (you can import the theme anywhere), and use it to reference the value you need. For example `theme.global.colors.brand` should return you the brand color :)

jfplataroti replied:
Great idea!! 👍

jfplataroti commented on packages/web/pages/profile/address/new.tsx:
```diff
@@ -16,11 +16,7 @@ export default function NewForm({ googleKey, requestOptions, labels }) {
-        requestOptions={requestOptions}
-        labels={labels}
-      />
+      <AddressForm labels={labels} />
```
google key is now added by default from env vars.

djoyal replied:
Nice!

djoyal commented on packages/web/components/addressField/addressField.customization.ts:
```diff
@@ -0,0 +1,41 @@
+  }),
+  valueContainer: (provided) => ({
+    ...provided,
+    backgroundColor: BRAND_COLORS.gray_2,
```
Instead of calling `BRAND_COLORS` directly, you can use `theme.global.colors.X` (E.G. `theme.global.colors.gray_2`). This way we can avoid exporting the `BRAND_COLORS` and rely on having our theme properly configured :)

Bonus: you can also use this for other things, like spacing. I'm not familiar with how that actually works (I don't know what `theme.global.spacing` returns), I just know it's available.

jfplataroti replied:
nice, yes having also the spacing units from the theme would be better.

djoyal commented on packages/web/components/addressForm/addressForm.tsx:
```diff
@@ -0,0 +1,35 @@
+  return (
+    <Form>
+      <FormField>
+        <AddressField placeholder="test" icon={<Search />} />
```
Can we change `test` to a fake address? Looks like this was a temporary value :)

djoyal reviewed (commented):
This looks great! The custom components in the picker are a nice touch.

I left a few small comments about using `BRAND_COLORS`, but they're nothing major.

jfplataroti commented:
@djoyal I updated how the address picker selects the colour and units base on the feedback. 

Now it uses `theme.global.colors` and also added a new object in global `theme.global.units` with numeric values. I couldn't use `spacing` or `sizes` because both are strings. 

Let me know if it makes sense.

djoyal commented:
@jfplataroti the changes look good to me, thanks for making them!
